cpp/src/parquet/encryption/file_key_wrapper.cc

```diff
@@ -86,9 +88,22 @@ std::string FileKeyWrapper::GetEncryptionKeyMetadata(const std::string& data_key
   // Internal key material storage: key metadata and key material are the same
   if (store_key_material_internally) {
     return serialized_key_material;
-  } else {
-    throw ParquetException("External key material store is not supported yet.");
   }
+  // External key material storage: key metadata is a reference to a key in the material
+  // store
+  if (key_id_in_file.empty()) {
+    if (is_footer_key) {
+      key_id_in_file = KeyMaterial::kFooterKeyIdInFile;
+    } else {
+      key_id_in_file =
+          KeyMaterial::kColumnKeyIdInFilePrefix + std::to_string(key_counter_);
+      key_counter_++;
```

Comment:
Why are we appending this integer? Who else needs to know about this ID? It seems like this detail deserves more comments.